### PR TITLE
Consumer Scheduling

### PR DESCRIPTION
**Pull request by kylehuntsman**
### Overview

Added scheduling logic for the consumer to consume messages during a configurable time window. Also added the moment library.

### CODE REVIEW

File: lib/sqs/sqs-base.js [hunk 1]
```diff
@@ -24,6 +24,14 @@ const DEFAULT_BASE_AWS_EXT_CONFIG = {
       sqsOptions: {
         region: 'us-west-2'
       },
+      consumer: {
+        scheduler: {
+          scheduled: false,
+          start: '00:00:00',
+          duration: 2 * 60 * 60, // 2hrs
```
Comment (sukrit007) [inline lib/sqs/sqs-base.js]: Suggestion:  Use human readable duration

File: lib/sqs/sqs-base.js [hunk 2]
```diff
@@ -24,6 +24,14 @@ const DEFAULT_BASE_AWS_EXT_CONFIG = {
...
+          maxWaitSeconds: 6 * 60 * 60 // 6hrs
```
Comment (sukrit007) [inline lib/sqs/sqs-base.js]: Should this be maxVisibilityTimeout ?  I did not see reference for maxWaitSeconds in the code

File: test/unit/sqs/sqs-consumer-spec.js [hunk 3]
```diff
@@ -244,10 +248,156 @@ describe('SqsConsumer', () => {
...
+        clock.restore();
+      });
+
+      it('resolves an empty array', () => {
+        sqs.receiveMessage.returns({ promise: () => Promise.resolve(messages)});
+        return sqs.receiveMessage()
+          .promise()
+          .then(data => consumer._scheduledConsuming(data))
+          .then(data => {
+            data.should.deep.equal({});
```
Comment (sukrit007) [inline test/unit/sqs/sqs-consumer-spec.js:327]: Suggestion: Would be great if we can also assert that message visibility is set to to required duration

**Review by sukrit007 — COMMENTED** [review]
@FunnyGopher I like your approach. Just added minor comments.  Let me know if you need clarifications with any of those

File: lib/sqs/sqs-base.js (hunk 2 above)
Comment (kylehuntsman) [reply lib/sqs/sqs-base.js]: Yes it should

File: lib/sqs/sqs-base.js (hunk 1 above)
Comment (kylehuntsman) [reply lib/sqs/sqs-base.js]: Added [`human-interval`](https://www.npmjs.com/package/human-interval) package from npm.

File: test/unit/sqs/sqs-consumer-spec.js [hunk 4]
```diff
@@ -244,10 +248,234 @@ describe('SqsConsumer', () => {
...
+
+    context('when consumer config is not present', () => {
+      before(() => {
+        delete config.defaults.consumer.scheduler;
+        consumer = new SqsConsumer({sqs: sqs}, msgBody => Promise.resolve(), {sqs: config});
+      });
+
+      it('resolves the original data', () => {
+        sqs.receiveMessage.returns({ promise: () => Promise.resolve(messages)});
+        return sqs.receiveMessage()
```
Comment (sukrit007) [inline test/unit/sqs/sqs-consumer-spec.js]: Not sure I follow this line.  Line 290 mocks the receiveMessage method and the test is calling mocked method receiveMessage.   
- Is there any reason we need to call receiveMessage when we are just testing _scheduledConsuming ?

Shouldn't the test be only calling consumer._scheduledConsuming(messages) ?

File: lib/sqs/sqs-base.js (hunk 2 above)
Comment (sukrit007) [reply lib/sqs/sqs-base.js]: LGTM!

File: lib/sqs/sqs-base.js (hunk 1 above)
Comment (sukrit007) [reply lib/sqs/sqs-base.js]: LGTM!

File: test/unit/sqs/sqs-consumer-spec.js (hunk 3 above)
Comment (sukrit007) [reply test/unit/sqs/sqs-consumer-spec.js:327]: LGTM!

File: test/unit/sqs/sqs-consumer-spec.js (hunk 4 above)
Comment (kylehuntsman) [reply test/unit/sqs/sqs-consumer-spec.js]: Totally agree. All the _scheduledConsuming tests actually did this. I'll fix them.

**Review by sukrit007 — APPROVED** [review]

File: test/unit/sqs/sqs-consumer-spec.js (hunk 4 above)
Comment (sukrit007) [reply test/unit/sqs/sqs-consumer-spec.js]: LGTM!

**Pull request merged**